spec/requests/messages_spec.rb

```diff
@@ -41,37 +41,83 @@
+
+      before do
+        post organization_messages_url(organization, as: user),
+             params: { message: msg_attrs, request_id: request.id, contributor_id: other_contributor.id }
+      end
+
+      it 'renders a not found' do
+        expect(response).to be_not_found
+      end
+    end
   end
 
   describe 'PATCH /message/:id' do
     let(:previous_text) { 'Previous text' }
-    let(:message) { create(:message, creator_id: user.id, text: previous_text, request: request) }
     let(:new_attrs) { { text: 'Grab your coat and get your hat' } }
-    subject { -> { patch organization_message_url(message.organization, message, as: user), params: { message: new_attrs } } }
 
-    it { should change { message.reload && message.text }.from(previous_text).to('Grab your coat and get your hat') }
+    context 'with message not from the organization prevents updating ' do
+      let(:message) { create(:message, creator_id: user.id, text: previous_text) }
 
-    it 'shows success notification' do
-      subject.call
-      expect(flash[:success]).not_to be_empty
-    end
+      subject { -> { patch organization_message_url(organization, message, as: user), params: { message: new_attrs } } }
 
-    it 'redirects to the conversation link the message belongs to' do
-      subject.call
-      expect(response).to redirect_to(message.chat_message_link)
+      it 'renders a notfound' do
+        subject.call
+        expect(response).to be_not_found
+      end
     end
 
-    context 'not manually created message' do
-      let(:message) { create(:message, creator_id: nil, request: request) }
+    context 'with a message of the organization' do
+      let(:message) { create(:message, creator_id: user.id, text: previous_text, request: request, sender: contributor) }
+
+      subject { -> { patch organization_message_url(message.organization, message, as: user), params: { message: new_attrs } } }
 
-      it 'does not update the requested message' do
+      it {
+        should change {
+                 message.reload.text
+               }.from(previous_text).to('Grab your coat and get your hat')
+      }
+
+      it 'shows success notification' do
         subject.call
-        expect(response).not_to be_successful
+
+        expect(flash[:success]).not_to be_empty
       end
 
-      it 'shows error notification' do
+      it 'redirects to the conversation link the message belongs to' do
         subject.call
-        expect(flash[:error]).not_to be_empty
+        expect(response).to redirect_to(message.chat_message_link)
+      end
+
+      context 'not manually created message' do
+        let(:message) { create(:message, creator_id: nil, request: request) }
+
+        it 'does not update the requested message' do
+          subject.call
+          expect(response).not_to be_successful
+        end
+
+        it 'shows error notification' do
+          subject.call
+          expect(flash[:error]).not_to be_empty
```

Comment:
I have been refactoring these in my other PR to make use of the new `CapybaraPage` helper you added. So instead of just checking that the flash is not empty, we can check that the page has content xyz
See https://github.com/tactilenews/100eyes/pull/1999/files#diff-71c5d728e921a45b367bd360e1bbeb7333547d2b23b5f1105ca6b72a941e3ec2R175-R177
Do you like that change? It's more explicit which I know you favor in other cases.

Reply:
Ok, it doesn't quite work like that because the `response.body` is the information that you are being redirected. I still updated the test to check that the flash success message contained the exact string, which contains the number of contributors I expect to receive the message based on belonging to the organization.